registry/storage/linkedblobstore.go

```diff
@@ -263,22 +265,25 @@ func (lbs *linkedBlobStore) Enumerate(ctx context.Context, ingestor func(digest.
 			if err == distribution.ErrBlobUnknown {
 				return nil
 			}
+			retErr = err
 			return err
 		}
 
 		err = ingestor(digest)
 		if err != nil {
+			retErr = err
 			return err
 		}
 
 		return nil
 	})
 
-	if err != nil {
-		return err
+	// We check retErr first, in case the Walk function somehow mutates the error
```

Comment:
If the error is mutated, wouldn't we still want the mutated error? I guess I am just trying to figure out if there is a good reason not to just `return lbs.driver.Walk(....`